cocotb/share/makefiles/Makefile.pylib.Msys

```diff
@@ -35,3 +35,13 @@ ifeq ($(PYTHON_LIBDIR),None)
 endif
 
 PYTHON_INCLUDEDIR := $(shell $(PYTHON_BIN) -c 'import distutils.sysconfig; print( distutils.sysconfig.get_python_inc() )')
+
+# On Windows there is no support for @rpath/LD_LIBRARY_PATH so we need to set PATH to find cocotb libraries
+# https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order?redirectedfrom=MSDN#standard-search-order-for-desktop-applications
+export PATH := $(LIB_DIR):$(PATH)
+
+# Set PYTHONHOME to properly populate sys.path in embedded python interpreter
+export PYTHONHOME := $(shell $(PYTHON_BIN) -c "from distutils.sysconfig import get_config_var; print(get_config_var('prefix'))")
+
+# For tcl scripts the path to dll has to be in form of C:/User/..
+export LIB_DIR := $(shell sh -c 'cd $(LIB_DIR) && pwd -W')
```

Comment:
I don't think these belong in this file - `pylib` is documented as:
```
# All common python related rules, kept only for downstream C extensions modules
# which copy the pattern of the endian_swapper HAL example. In future these modules
# should use `setuptools` to build their HAL interface
```

Reply:
```suggestion

# On Windows there is no support for @rpath/LD_LIBRARY_PATH so we need to set PATH to find cocotb libraries
# https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order?redirectedfrom=MSDN#standard-search-order-for-desktop-applications
export PATH := $(LIB_DIR):$(PATH)

# Set PYTHONHOME to properly populate sys.path in embedded python interpreter
export PYTHONHOME := $(shell $(PYTHON_BIN) -c "from distutils.sysconfig import get_config_var; print(get_config_var('prefix'))")

# For tcl scripts the path to dll has to be in form of C:/User/..
export LIB_DIR := $(shell cygpath -m "$(LIB_DIR)")
```

Reply:
I was thinking to avoid `cygpath` (no good reason) as long as possible but any is OK for me.

Reply:
I'd argue `cygpath` is more explicit here, as well as behaving predictably when paths do not exist.